lib/medtronic/processData.js

```diff
@@ -472,69 +567,101 @@ function buildBasalProfiles(records) {
   };
 
   // we start with the current basal profiles and work our way back
-  var basalProfiles = filterHistory([RECORD_TYPES.CHANGE_BASAL_PROFILE_OLD,
+  var changes = filterHistory([RECORD_TYPES.CHANGE_BASAL_PROFILE_OLD,
                                  RECORD_TYPES.CHANGE_BASAL_PROFILE_NEW,
-                                 RECORD_TYPES.SELECT_BASAL_PROFILE], records).reverse();
+                                 RECORD_TYPES.SELECT_BASAL_PROFILE,
+                                 RECORD_TYPES.BOLUS_WIZARD_CHANGE_OLD], records).reverse();
   var postrecords = [];
-  var scheduleChanges = [];
-  var activeSchedule, prevRecord;
+  var settingsChanges = [];
+  var prevDate = null;
 
   // when we record schedule changes, we need to start with the
   // most recent schedules and work our way back
-  var scheduleChange = cfg.settings.basalSchedules;
-  scheduleChanges.push(_.clone(scheduleChange));
+  var settingsChange = {
+    basalSchedules : _.clone(cfg.settings.basalSchedules),
+    bgTarget : _.clone(cfg.settings.bgTarget),
+    insulinSensitivity : _.clone(cfg.settings.insulinSensitivity),
+    carbRatio : _.clone(cfg.settings.carbRatio),
+    units : _.clone(cfg.settings.units),
+    activeSchedule : _.clone(cfg.settings.activeSchedule)
+  };
+  settingsChanges.push(_.cloneDeep(settingsChange));
```

Comment:
Oh wait, now I remember! It's because I modify `settingsChange` on each pass, so if you set `settingsChange = cfg.settings`, you end up modifying `cfg.settings` too.